src/bun.js/bindings/webcore/JSFetchHeaders.cpp

```diff
@@ -450,7 +450,7 @@ static inline JSC::EncodedJSValue jsFetchHeadersPrototypeFunction_toJSONBody(JSC
         for (auto it = vec.begin(); it != vec.end(); ++it) {
             auto& name = it->key;
             auto& value = it->value;
-            obj->putDirect(vm, Identifier::fromString(vm, name), jsString(vm, value), 0);
+            obj->putDirect(vm, Identifier::fromString(vm, name.convertToASCIILowercase()), jsString(vm, value), 0);
```

Comment:
I think this is right because we should only be allowing ASCII header names as mentioned in the RFC for both http/1.1 and http2
https://www.rfc-editor.org/rfc/rfc7540#section-1:~:text=Just%20as%20in%20HTTP/1.x%2C%20header%20field%20names%20are%20strings%20of%20ASCII%0A%20%20%20characters%20that%20are%20compared%20in%20a%20case%2Dinsensitive%20fashion